### DISCUSSION TITLE
Visitor Count

### DISCUSSION THREAD

**Discussion by SolarDiode** [RaspAP/raspap-webgui#1260] [Q&A]
I have RaspAP and NDS working perfectly, just how I want them, without an internet connection.
I would like to observe the number of connections that the AP attracts. Is this better done within NDS or can the AP do this?

Thanks

**Comment by billz** [comment]
Yes, this can be done. Broadly, you have two options here:

1) Simplest is to use RaspAP's active DHCP leases  on the **DHCP Server > Client list** page. This includes client MAC address, IP, etc. Effectively, this represents a count of unique active visitors (devices) connected to the AP.  
2) Uninstall NDS and install OpenNDS (a fork of the original code) then enable the [dynamic user login page](https://opennds.readthedocs.io/en/stable/customize.html#pre-installed-dynamic-user-email-login-page-sequence). This generates a login page asking for a username and email address which are then logged to the filesystem. This differs from option 1 in that it gives you a history of all unique users, rather than devices which expire according to your DHCP server's settings (default is 12 hours).

The latter is more involved but potentially gives you more useful info, depending on your specific goals.
